docs/internals/development.rst

```diff
@@ -0,0 +1,55 @@
+Developer Information
+=====================
+
+The following information is intended for developers who wish to modify the
+code.
+
+The pages on :doc:`testing` and :doc:`documentation` are also relevant.
+
+Building BridgeStan
+-------------------
+
+Developers should follow the instructions in :doc:`../getting-started` to ensure
+they have the minimal C++ build tools required for working with BridgeStan.
+
+C++
+---
+
+* We use the C++1y standard for compilation (``-std=c++1y`` in both clang and gcc). This is partway between C++11 and C++14, and is what Stan requires.
```

Comment:
As of R 4.3, they support C++17.  It rolled out earlier this year with a late 4.2.x.  See https://developer.r-project.org/blosxom.cgi/R-devel/NEWS/2023/01/27#n2023-01-27

I don't know if Python is a blocker.

Reply:
Because we don't rely on any specific C++ ABI, we could use any C++ standard we like for this project. This comment is just there because it's what our makefiles pass and what the rest of Stan uses